system/training.py

```diff
@@ -91,125 +98,168 @@ def filter_groups(param_groups, groups):
+            for b_idx, b in enumerate(batch_data):
+                args.append([get_params(opt_state), b[0], b[1], b_idx])
 
-        batch_dp = np.zeros_like(init_params)
+            results = pool.map(run_simulation, args)
 
-        for grads, preds, labels in results:
-            batch_dp += grads
-            epoch_predictions.append(preds)
-            epoch_labels.append(labels)
+            batch_dp = np.zeros_like(init_params)
 
-        count += 1
-        opt_state = opt_update(count, batch_dp, opt_state)
+            for grads, preds, labels in results:
+                batch_dp += grads
+                epoch_predictions.append(preds)
+                epoch_labels.append(labels)    
 
-    epoch_predictions = np.array(epoch_predictions)
-    epoch_labels = np.array(epoch_labels)
+            count += 1
+            opt_state = opt_update(count, batch_dp, opt_state)
 
-    np.savez("run_"+str(epoch)+".npz", preds=epoch_predictions, labels=epoch_labels, filenames=fn, params=get_params(opt_state))
+        epoch_predictions = np.array(epoch_predictions)
+        epoch_labels = np.array(epoch_labels)
+
+        np.savez("run_{}.npz".format(epoch), preds=epoch_predictions, labels=epoch_labels, filenames=epoch_filenames, params=get_params(opt_state))
+    
+        mae = np.mean(np.abs(epoch_predictions-epoch_labels))
+        mean = np.mean(epoch_predictions-epoch_labels)
+        if len(epoch_predictions) > 1:
+            pearson_r = stats.pearsonr(epoch_predictions, epoch_labels)
+            r2_score = sklearn.metrics.r2_score(epoch_predictions, epoch_labels)
+            print('''
+
+Epoch: {}
+-----------------
+Pearson R: {}
+R2 score: {}
+MAE: {}
+Mean: {}
+-----------------
+
+            '''.format(epoch,pearson_r[0], r2_score, mae, mean))
+        else:
+            print('''
+            
+Epoch: {}
+-----------------
+MAE: {}
+Mean: {}
+-----------------
+            
+            '''.format(epoch,mae, mean))
+        
+    return preds, labels, get_params(opt_state)
+
+def initialize_optimizer(optimizer, lr):
+    
+    if optimizer == 'Adam':
+        opt_init, opt_update, get_params = optimizers.adam(lr)
+    elif optimizer == 'SGD':
+        opt_init, opt_update, get_params = optimizers.sgd(lr)
+    elif optimizer == 'RMSProp':
+        opt_init, opt_update, get_params = optimizers.rmsprop(lr)
+    elif optimizer == 'Adagrad':
+        opt_init, opt_update, get_params = optimizers.adagrad(lr)
+    elif optimizer == 'SM3':
+        opt_init, opt_update, get_params = optimizers.sm3(lr)
+        
+    return opt_init, opt_update, get_params
+
+if __name__ == "__main__":
+    
+    config_file = glob.glob('*.json')
+    if len(config_file) == 0:
+        raise Exception('config file not found')
+    elif len(config_file) > 1:
+        raise Exception('multiple config files found')
+    with open(config_file[0], 'r') as file:
+        config = json.load(file)
+   
+    globals()['properties'] = config
```

Comment:
don't dump this into globals(). This part of the code has global visibility anyways so simply doing ```properties = config``` should work